python/cudf/cudf/core/series.py

```diff
@@ -3531,6 +3531,24 @@ def one_hot_encoding(self, cats, dtype="float64"):
             cats = pd.Series(cats, dtype="object")
         dtype = cudf.dtype(dtype)
 
+        if cats is not None:
+            try:
+                cats_col = as_column(cats, nan_as_null=False, dtype=self.dtype)
+            except TypeError:
+                raise ValueError("Cannot convert `cats` as cudf column.")
+
+            if self._column.size * cats_col.size >= np.iinfo("int32").max:
+                raise ValueError(
+                    "Size limitation exceeded: series.size * category.size < "
+                    "np.iinfo('int32').max. Consider reducing size of category"
+                )
+
+            res = libcudf.transform.one_hot_encode(self._column, cats_col)
+            if dtype.type == np.bool_:
+                return list(res.values())
```

Comment:
What is this case handling?

Reply:
By default libcudf returns boolean columns so we don't need to cast the columns here.